File: 55-jump-game/jump-game.js
var canJump = function(nums) {
    var far =0
    for (var i=0;i<nums.length;i++){
        if (i>far){
            return false
        }
        far=Math.max(far, i+ nums[i])
        if (far >= nums.length-1){
            return true
        }
    }

    
};

File: 55-jump-game/jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        far=0
        for i in range(len(nums)):
            if i > far:
                return False
            
            far = max(i+nums[i],far)
        return far >= len(nums) - 1

            

        